src/rpc/blockchain.cpp

```diff
@@ -1211,46 +1211,48 @@ static void BuriedForkDescPushBack(UniValue& softforks, const std::string &name,
     // A timeout value of 0 guarantees a softfork will never be activated.
     // This is used when merging logic to implement a proposed softfork without a specified deployment schedule.
-    if (consensusParams.vDeployments[id].nTimeout <= 1230768000) return;
+    if (consensusParams.vDeployments[id].IsNeverActive()) return;
 
-    UniValue bip9(UniValue::VOBJ);
+    UniValue details(UniValue::VOBJ);
     const ThresholdState thresholdState = VersionBitsTipState(consensusParams, id);
     switch (thresholdState) {
-    case ThresholdState::DEFINED: bip9.pushKV("status", "defined"); break;
-    case ThresholdState::STARTED: bip9.pushKV("status", "started"); break;
-    case ThresholdState::LOCKED_IN: bip9.pushKV("status", "locked_in"); break;
-    case ThresholdState::ACTIVE: bip9.pushKV("status", "active"); break;
-    case ThresholdState::FAILED: bip9.pushKV("status", "failed"); break;
-    }
-    if (ThresholdState::STARTED == thresholdState)
-    {
-        bip9.pushKV("bit", consensusParams.vDeployments[id].bit);
-    }
-    bip9.pushKV("start_time", consensusParams.vDeployments[id].nStartTime);
-    bip9.pushKV("timeout", consensusParams.vDeployments[id].nTimeout);
+    case ThresholdState::DEFINED: details.pushKV("status", "defined"); break;
+    case ThresholdState::STARTED: details.pushKV("status", "started"); break;
+    case ThresholdState::MUST_SIGNAL: details.pushKV("status", "must_signal"); break;
+    case ThresholdState::LOCKED_IN: details.pushKV("status", "locked_in"); break;
+    case ThresholdState::ACTIVE: details.pushKV("status", "active"); break;
+    case ThresholdState::FAILED: details.pushKV("status", "failed"); break;
+    }
+    if (ThresholdState::STARTED == thresholdState || ThresholdState::MUST_SIGNAL == thresholdState || ThresholdState::LOCKED_IN == thresholdState) {
+        details.pushKV("bit", consensusParams.vDeployments[id].bit);
+    }
+    details.pushKV("startheight", consensusParams.vDeployments[id].startheight);
+    details.pushKV("timeoutheight", consensusParams.vDeployments[id].timeoutheight);
+    details.pushKV("lockinontimeout", consensusParams.vDeployments[id].lockinontimeout);
     int64_t since_height = VersionBitsTipStateSinceHeight(consensusParams, id);
-    bip9.pushKV("since", since_height);
-    if (ThresholdState::STARTED == thresholdState)
-    {
+    details.pushKV("since", since_height);
+    if (ThresholdState::STARTED == thresholdState || ThresholdState::MUST_SIGNAL == thresholdState || ThresholdState::LOCKED_IN == thresholdState) {
         UniValue statsUV(UniValue::VOBJ);
-        BIP9Stats statsStruct = VersionBitsTipStatistics(consensusParams, id);
+        VBitsStats statsStruct = VersionBitsTipStatistics(consensusParams, id);
         statsUV.pushKV("period", statsStruct.period);
-        statsUV.pushKV("threshold", statsStruct.threshold);
+        if (ThresholdState::STARTED == thresholdState) {
+            statsUV.pushKV("threshold", statsStruct.threshold);
```

Comment:
Test should be `thresholdState != ThresholdState::LOCKED_IN` -- now that https://github.com/bitcoin/bips/pull/1021/ is merged, threshold is still relevant during MUST_SIGNAL phase.